spring-cloud-dataflow-docs/src/main/asciidoc/streams.adoc

```diff
@@ -82,6 +82,68 @@ The xref:getting-started#getting-started[Getting Started] section shows you how
 
 Note that the shell calls the Data Flow Servers' REST API. For more information on making HTTP requests directly to the server, consult the <<api-guide, REST API Guide>>.
 
+=== Stream Application DSL
+
+The Stream Pipeline DSL described in the previous section automatically sets the input and output binding properties of each Spring Cloud Stream application.
+This can be done because there is only one input and/or output destination in a Spring Cloud Stream application that uses the provided binding interface of a `Source`, `Processor`, or `Sink`.
+However, a Spring Cloud Stream application can define a custom binding interface such as the one shown below
+
+[source,java]
+----
+public interface Barista {
+
+    @Input
+    SubscribableChannel orders();
+
+    @Output
+    MessageChannel hotDrinks();
+
+    @Output
+    MessageChannel coldDrinks();
+}
+----
+
+or as is common when creating a Kafka Streams application,
+
+[source,java]
+----
+interface KStreamKTableBinding {
+
+    @Input
+    KStream<?, ?> inputStream();
+
+    @Input
+    KTable<?, ?> inputTable();
+}
+----
+
+In these cases with multiple input and output bindings, Data Flow can not make any assumptions about the flow of data from one application to another.
+Therefore the developer needs to set the binding properties to 'wire up' the application.
+The *Stream Application DSL* uses a `comma`, instead of the `pipe symbol`, to indicate that Data Flow should not configure the binding properties of the application.
+For example:
+
+`dataflow:> stream create --definition "orderGeneratorApp, baristaApp, hotDrinkDeliveryApp, coldDrinkDeliveryApp" --name myCafeStream`
+
+There are four applications in this stream.
+ The baristaApp has two output destinations, `hotDrinks` and `coldDrinks` intended to be consumed by the `hotDrinkDeliveryApp` and `coldDrinkDeliveryApp` respectively.
+When deploying this stream, you need to set the binding properties so that the `baristaApp` sends hot drink messages to the `hotDrinkDeliveryApp` and cold drink mesage to the `coldDrinkDeliveryApp`.
```

Comment:
> When deploying this stream, you need to set the binding properties so that the `baristaApp` sends hot drink messages to the `hotDrinkDeliveryApp` and cold drink mesage to the `coldDrinkDeliveryApp`.

Perhaps:

> When deploying this stream, you need to set the binding properties so that the `baristaApp` sends hot drink messages to the `hotDrinkDeliveryApp` and cold drink messages to the `coldDrinkDeliveryApp` topics respectively.

Reply:
topics -> destinations